firefox-ios/firefox-ios-tests/Tests/ClientTests/MainMenu/MainMenuMiddlewareTests.swift

```diff
@@ -8,37 +8,699 @@ import XCTest
 
 @testable import Client
 
-final class MainMenuMiddlewareTests: XCTestCase {
+final class MainMenuMiddlewareTests: XCTestCase, StoreTestUtility {
+    var mockGleanWrapper: MockGleanWrapper!
+    var mockStore: MockStoreForMiddleware<AppState>!
+
     override func setUp() {
         super.setUp()
+        mockGleanWrapper = MockGleanWrapper()
         // Due to changes allow certain custom pings to implement their own opt-out
         // independent of Glean, custom pings may need to be registered manually in
         // tests in order to put them in a state in which they can collect data.
         Glean.shared.registerPings(GleanMetrics.Pings.shared)
         Glean.shared.resetGlean(clearStores: true)
```

Comment:
if we're using the `mockGleanWrapper`, can we remove using the shared instances of Glean?